ASM/c/kaleido_scope.c

```diff
@@ -0,0 +1,241 @@
+            func_800F64E0(0);
+            // returns to title
+            //SEQCMD_STOP_SEQUENCE(0, 10);
+            //SEQCMD_STOP_SEQUENCE(3, 10);
+            play->pause_ctxt.state = PAUSE_STATE_REUTRN_TO_TITLE;
+        }
+    }
+}
+
+void kaleidoScope_Case20(z64_game_t* play) {
+    z64_pause_ctxt_t* pauseCtx = &play->pause_ctxt;
+
+    (*D_8082B25C_RELOCATED)--;
+    if ((*D_8082B25C_RELOCATED) == 0) {
+        play->pause_ctxt.state = PAUSE_STATE_RETURN_TO_TITLE_PROMPT;
+    } else if (((*D_8082B25C_RELOCATED) <= 80) &&
+               (play->common.input[0].pad_pressed.a || play->common.input[0].pad_pressed.s)) {
+        play->pause_ctxt.state = PAUSE_STATE_RETURN_TO_TITLE_PROMPT;
+        //func_800F64E0(0);
+    }
+}
+
+//void KaleidoScopeCall_Draw(z64_game_t* play) {
+//    KaleidoMgrOverlay* kaleidoScopeOvl = &gKaleidoMgrOverlayTable[0];
+//
+//    if (R_PAUSE_BG_PRERENDER_STATE >= PAUSE_BG_PRERENDER_READY) {
+//        if (((play->pause_ctxt.state >= PAUSE_STATE_OPENING_1) && (play->pause_ctxt.state <= PAUSE_STATE_SAVE_PROMPT)) ||
+//            ((play->pause_ctxt.state == PAUSE_STATE_RETURN_TO_TITLE_PROMPT) || (play->pause_ctxt.state == PAUSE_STATE_RETURN_TO_TITLE_PROMPT_TIMER)) ||
+//            ((play->pause_ctxt.state >= PAUSE_STATE_11) && (play->pause_ctxt.state <= PAUSE_STATE_CLOSING))) {
+//            if (gKaleidoMgrCurOvl == kaleidoScopeOvl) {
+//                sKaleidoScopeDrawFunc(play);
+//            }
+//        }
+//    }
+//}
+
+static s16 D_8082A6E0[] = { 100, 255 };
+
+void KaleidoScope_UpdatePrompt(z64_game_t* play) {
+    s8 stickAdjX = play->common.input[0].adjusted_x;
+    s16 step;
+
+    if (((play->pause_ctxt.state == PAUSE_STATE_SAVE_PROMPT) && (play->pause_ctxt.unk_1EC == 1)) ||
+        (play->pause_ctxt.state == PAUSE_STATE_14) || (play->pause_ctxt.state == PAUSE_STATE_16) || (play->pause_ctxt.state == PAUSE_STATE_RETURN_TO_TITLE_PROMPT)) {
+        if ((play->pause_ctxt.prompt_choice == 0) && (stickAdjX >= 30)) {
+            Audio_PlaySfxGeneral(NA_SE_SY_CURSOR, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
+                                 &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
+            play->pause_ctxt.prompt_choice = 4;
+        } else if ((play->pause_ctxt.prompt_choice != 0) && (stickAdjX <= -30)) {
+            Audio_PlaySfxGeneral(NA_SE_SY_CURSOR, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
+                                 &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
+            play->pause_ctxt.prompt_choice = 0;
+        }
+
+        step = ABS(VREG(61) - D_8082A6E0[VREG(62)]) / VREG(63);
+        if (VREG(61) >= D_8082A6E0[VREG(62)]) {
+            VREG(61) -= step;
+        } else {
+            VREG(61) += step;
+        }
+
+        VREG(63)--;
+        if (VREG(63) == 0) {
+            VREG(61) = D_8082A6E0[VREG(62)];
+            VREG(63) = VREG(60) + VREG(62);
+            VREG(62) ^= 1;
+        }
+    }
+}
```

Comment:
Your editor should have an option for automatically adding a new line at the end of files.